### PR TITLE
Hard-code .zmetadata in FSStore._normalize_key

### PR DESCRIPTION
**Pull request by joshmoore**
The normalization code in FSStore uses a list of well-known files to prevent incorrectly re-writing keys when dimension separator is "/" rather than ".". This was initialized only with the names specified in the spec, which left out ".zmetadata" from consolidated metadata.

see: #1121

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by joshmoore** [comment]
It would be good to have a failing test in the suite before this gets merged.

File: zarr/storage.py [hunk 1]
```diff
@@ -1378,7 +1378,12 @@ def _normalize_key(self, key):
         if key:
             *bits, end = key.split('/')
 
-            if end not in (self._array_meta_key, self._group_meta_key, self._attrs_key):
+            if end not in (
+                self._array_meta_key,
+                self._group_meta_key,
+                self._attrs_key,
+                ".zmetadata",  # see: #1121
```
Comment (d-v-b) [inline zarr/storage.py:1385]: could we do this via a class attribute like `_consolidated_meta_key`?

**Comment by joshmoore** [comment]
Any thoughts about getting this into the 2.17.x line?

**Comment by jhamman** [comment]
I'm going to close this as stale. Folks should feel free to reopen if there is interest in continuing this work.

**Pull request closed without merging**